### PR TITLE
Rewrite of `main.py` - Make it class based.

### PR DESCRIPTION
**Pull request by nicdgonzalez**
(This is my first pull request. My apologies if this is not done correctly.)

Remaining:
    + Create a new cog for the commands that were here previously.
        I couldn't get it to work where the commands would register correctly with a class based Bot instance. Which in my opinion, is not that big of a deal.

Commands:
    - logout
    - _eval
    - dbbackup

DISCLAIMER:
I did not test this. I don't have access to your database. Although I did not test it, I tried to use your exact same code just converted into a class based version.

Why I wrote this:
I watched you guys for a good while when I started learning how to make a Discord bot. You guys write great code. I just feel that creating a script style bot is not very pretty and it is just good practice to create classes to group similar code together.

Hopefully you guys end up using this. If not, that's cool too.

Have a good one!

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,228 +1,172 @@
...
+                    prefix = self.DEFAULTPREFIX
+                else:
+                    prefix = data['prefix']
+
+                content = 'My prefix here is `%s`' % (prefix)
+                await message.channel.send(content, delete_after=15)
+
+        await self.process_commands(message)
+
+    def mongo(self, collection):
```
Comment (Skelmis) [inline main.py:91]: If you are going to introduce a change in the way database documents are interacted with, please do so throughout the whole bot.

I am also not sure how I feel about this change

File: main.py [hunk 2]
```diff
@@ -1,228 +1,172 @@
...
+        """Solution for needing to restart the bot each time a new collection
+        needs to be created.
+
+        This way you just call this method and create the instance
+        of the collection when you need it in the Cog.
```
Comment (Skelmis) [inline main.py:96]: Also not entirely true, but it works so don't worry

**Review by Skelmis — COMMENTED** [review]
> I couldn't get it to work where the commands would register correctly with a class based Bot instance. Which in my opinion, is not that big of a deal.
This is just a dpy thing

Also, this cannot be considered a complete pull request until you add the commands n such. It would also be appreciated if you tested to the best of your ability.

**Comment by nicdgonzalez** [comment]
> > I couldn't get it to work where the commands would register correctly with a class based Bot instance. Which in my opinion, is not that big of a deal.
> 
> This is just a dpy thing
> 
> 
> 
> Also, this cannot be considered a complete pull request until you add the commands n such. It would also be appreciated if you tested to the best of your ability.

As far as testing goes, I know it works because I've created something extremely similar for my bots.

But I will log in with the bot and test it sometime next week and make sure that all is in working order. I'll let you know if I have any additional questions.

I only rewrote main.py due to a time constraint but next week I should be able to rewrite the rest of the modules.

If this is not really something you're interested in me completing, just let me know. Else, if I don't hear from you I'll assume you are interested.

Thanks for the response!

**Comment by Skelmis** [comment]
Safe as, I am somewhat refactoring stuff now. You may need to rebase / reflect those changes in your pull request

**Pull request closed without merging**